### PR TITLE
Add bull profit threshold

### PR DESCRIPTION
**Pull request by KMKoushik**
fixes ENG-1097

### CODE REVIEW

File: packages/frontend/src/state/bull/atoms.ts [hunk 1]
```diff
@@ -24,3 +26,36 @@ export const bullEthPnlPerctAtom = atom(new BigNumber(0))
...
 export const bullPositionLoadedAtom = atom(false)
+
+export const eulerUsdcBorrowRateAtom = atom(new BigNumber(0.05))
+export const eulerETHLendRateAtom = atom(new BigNumber(0.1))
+
+export const bullEulerUSDCDebtAtom = atom((get) => {
+  return get(bullEulerUsdcDebtPerShareAtom).times(get(bullSupplyAtom))
+})
+
+export const bullCurrentFundingAtom = atom((get) => {
```
Comment (KMKoushik) [inline packages/frontend/src/state/bull/atoms.ts:37]: @aleone

File: packages/frontend/src/state/bull/atoms.ts [hunk 2]
```diff
@@ -24,3 +26,36 @@ export const bullEthPnlPerctAtom = atom(new BigNumber(0))
...
+  const ethInterest = ethCollatInUsd.times(Math.exp(ethRate.div(365 / 2).toNumber())).minus(ethCollatInUsd)
+  const totalRate = ethInterest.minus(usdInterest)
+  const interestFunding = totalRate.div(crabShares).div(2)
+  const squeethFunding = get(currentImpliedFundingAtom)
+
+  return interestFunding.plus(squeethFunding).toNumber()
+})
+
+export const bullThresholdAtom = atom((get) => {
+  const funding = get(bullCurrentFundingAtom)
```
Comment (KMKoushik) [inline packages/frontend/src/state/bull/atoms.ts:56]: @aleone

File: packages/frontend/src/state/bull/utils.ts [hunk 3]
```diff
@@ -70,3 +70,8 @@ const calcCrabToMint = (depositEthAmount: BigNumber, strategyCollat: BigNumber,
 
   return totalSupply.times(depositorShare).div(new BigNumber(1).minus(depositorShare))
 }
+
+// https://docs.euler.finance/developers/numeric-limits#interestrate
+export const getEulerInterestRate = (apy: BigNumber) => {
```
Comment (KMKoushik) [inline packages/frontend/src/state/bull/utils.ts:75]: @aleone

**Review by aleone — COMMENTED** [review]
Some small comments.  Not sure we are calculating the ETH rate correctly as we need the lending rate, which is probably a function of utilization.

File: packages/frontend/src/state/bull/atoms.ts [hunk 4]
```diff
@@ -24,3 +26,36 @@ export const bullEthPnlPerctAtom = atom(new BigNumber(0))
...
+  const ethPrice = get(indexAtom).sqrt()
+  const usdRate = get(eulerUsdcBorrowRateAtom)
+  const ethRate = get(eulerETHLendRateAtom)
+  const usdDebt = get(bullEulerUSDCDebtAtom)
+  const ethCollat = get(bullDepositedEthInEulerAtom)
+  const ethCollatInUsd = ethCollat.times(ethPrice)
+  const crabShares = get(bullCrabBalanceAtom).times(get(crabUSDValueAtom))
+
+  const usdInterest = usdDebt.times(Math.exp(usdRate.div(365 / 2).toNumber())).minus(usdDebt)
+  const ethInterest = ethCollatInUsd.times(Math.exp(ethRate.div(365 / 2).toNumber())).minus(ethCollatInUsd)
```
[the remark this answers is not in the document]
Comment (KMKoushik) [reply packages/frontend/src/state/bull/atoms.ts:47]: We are taking appropriate interest rate from euler contract itself https://github.com/euler-xyz/euler-contracts/blob/fa9398728165676a5666939d8c34a7578d8e1919/contracts/views/EulerSimpleLens.sol#L78

ie)
WETH - supplyApy
USDC - borrowApy

File: packages/frontend/src/state/bull/utils.ts (hunk 3 above)
Comment (KMKoushik) [reply packages/frontend/src/state/bull/utils.ts:75]: Already mentioned in the previous comment. We are using the appropriate APY from Euler contracts

File: packages/frontend/src/state/bull/atoms.ts [hunk 5]
```diff
@@ -24,3 +26,36 @@ export const bullEthPnlPerctAtom = atom(new BigNumber(0))
...
+  const totalRate = ethInterest.minus(usdInterest)
+  const interestFunding = totalRate.div(crabShares).div(2)
```
[the remark this answers is not in the document]
Comment (aleone) [reply packages/frontend/src/state/bull/atoms.ts:49]: this is correct, disregard, discussed with KM

**Pull request merged**